src/Database/LSMTree/Internal/Config.hs

```diff
@@ -276,39 +275,6 @@ bloomFilterAllocForLevel conf (LevelNo l) =
                                       0 -> Just x
                                       _ -> r (k-1)) (const Nothing) xs n
 
-{-------------------------------------------------------------------------------
-  Fence pointer index
--------------------------------------------------------------------------------}
-
--- | Configure the type of fence pointer index.
---
--- TODO: this configuration option currently has no effect: 'CompactIndex' is
--- always used.
-data FencePointerIndex =
-    -- | Use a compact fence pointer index.
-    --
-    -- The compact index type is designed to work with keys that are large
-    -- cryptographic hashes, e.g. 32 bytes.
-    --
-    -- When using the 'IndexCompact', additional constraints apply to the
-    -- 'Database.LSMTree.Internal.Serialise.Class.serialiseKey' function. The
-    -- __Minimal size__ law should be satisfied:
-    --
-    -- [Minimal size] @'Database.LSMTree.Internal.RawBytes.size'
-    --   ('Database.LSMTree.Internal.Serialise.Class.serialiseKey' x) >= 8@
-    --
-    -- Use 'Database.LSMTree.Internal.Serialise.Class.serialiseKeyMinimalSize'
-    -- to test this law.
-    CompactIndex
-    -- | Use an ordinary fence pointer index, without any constraints on
-    -- serialised keys.
-  | OrdinaryIndex
-  deriving stock (Show, Eq)
-
-instance NFData FencePointerIndex where
-  rnf CompactIndex  = ()
-  rnf OrdinaryIndex = ()
-
```

Comment:
I'd prefer to keep this as its own config option, and then have a small `FencePointerIndex -> IndexType` translation here like we have `bloomFilterAllocForLevel` for the bloom filter allocation option.  If we do this, it means the index config option and index type flag can then change independently in the future.

```haskell
{-------------------------------------------------------------------------------
  Fence pointer index
-------------------------------------------------------------------------------}

-- | Configure the type of fence pointer index.
--
-- TODO: this configuration option currently has no effect: 'CompactIndex' is
-- always used.
data FencePointerIndex =
    -- | Use a compact fence pointer index.
    --
    -- The compact index type is designed to work with keys that are large
    -- cryptographic hashes, e.g. 32 bytes.
    --
    -- When using the 'IndexCompact', additional constraints apply to the
    -- 'Database.LSMTree.Internal.Serialise.Class.serialiseKey' function. The
    -- __Minimal size__ law should be satisfied:
    --
    -- [Minimal size] @'Database.LSMTree.Internal.RawBytes.size'
    --   ('Database.LSMTree.Internal.Serialise.Class.serialiseKey' x) >= 8@
    --
    -- Use 'Database.LSMTree.Internal.Serialise.Class.serialiseKeyMinimalSize'
    -- to test this law.
    CompactIndex
    -- | Use an ordinary fence pointer index, without any constraints on
    -- serialised keys.
  | OrdinaryIndex
  deriving stock (Show, Eq)

instance NFData FencePointerIndex where
  rnf CompactIndex  = ()
  rnf OrdinaryIndex = ()
  
indexTypeForRun :: FencePointerIndex -> Indextype
indexTypeForRun = ...
```